phoenix-core/src/it/java/org/apache/phoenix/end2end/index/ViewIndexIT.java

```diff
@@ -458,6 +457,124 @@ public void testMultiTenantViewGlobalIndex() throws Exception {
+                + "DOUBLE2 DECIMAL(12, 3), DOUBLE3 DECIMAL(12, 3), TEXT2 VARCHAR, TEXT4 VARCHAR, EMAIL1 VARCHAR, PHONE1 CHAR(10), URL1 VARCHAR "
+                + "CONSTRAINT PKVIEW PRIMARY KEY (%s)) AS SELECT * FROM %s WHERE KEY_PREFIX = '%s'", fullViewName,viewPkColumns, fullTableName, keyPrefix);
+        String indexDdl = String.format("CREATE INDEX IF NOT EXISTS %s "
+                + "ON %s (%s)\n"
+                + "INCLUDE (DOUBLE3, DOUBLE2, DATE_TIME3, TEXT2, URL1, INT3"
+                + "  )", fullIndexName, fullViewName, indexPKColumns);
+
+        conn.createStatement().execute(tableDdl);
+        conn.createStatement().execute(viewDdl);
+        conn.createStatement().execute(indexDdl);
+
+        String childViewName = String.format("S_%s.\"%s\"", generateUniqueName(), keyPrefix);
+        String viewChildDDl = String.format("CREATE VIEW IF NOT EXISTS %s AS SELECT * FROM %s", childViewName, fullViewName);
+        Connection conn2 = null;
+        if (isNamespaceMapped) {
+            conn2 = getTenantConnection(TENANT1);
+        } else {
+            conn2 = conn;
+        }
+        conn2.createStatement().execute(viewChildDDl);
+
+        // We picked this number because it ends with the separator byte.
+        int int1= 1792;
+        String upsert = "UPSERT INTO " + childViewName +
+                "(DATE_TIME1, INT1, TEXT1, TEXT2, DOUBLE1, DATE_TIME2, DATE_TIME3, INT3, DOUBLE2, DOUBLE3, "
+                + "TEXT3, TEXT4, EMAIL1, PHONE1, URL1, INT2, LAST_UPDATE"
+                + (isNamespaceMapped? ") " : ",ORGANIZATION_ID) ")
+                + " VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?" + (isNamespaceMapped? ")":",?)");
+        PreparedStatement upsertStmt = conn2.prepareStatement(upsert);
+        Date date =  Date.valueOf("2019-02-17");
+        upsertStmt.setDate(1, date);
+        upsertStmt.setInt(2, int1);
+        upsertStmt.setString(3, "text1");
+        upsertStmt.setString(4, "text2"); // TEXT2
+        upsertStmt.setDouble(5,254.564);
+        upsertStmt.setDate(6, null); //DATE_TIME2
+        upsertStmt.setDate(7, Date.valueOf("2019-07-16"));
+        upsertStmt.setInt(8, int1);
+        upsertStmt.setDouble(9, 4.09);
+        upsertStmt.setDouble(10, 0.249);
+        upsertStmt.setString(11,"text3"); // TEXT3
+        upsertStmt.setString(12, null); // TEXT4
+        upsertStmt.setString(13,"VScZBIjkO3QyUCMtkUEgmvL9xH0KJjwKi1gpxRv1ghonWcUMoksTWFKR4SD2yUg9@gmail.com");
+        upsertStmt.setString(14, null);
+        upsertStmt.setString(15, "https://www.sssssss.com");
+        upsertStmt.setNull(16, Types.BIGINT); //INT2
+        upsertStmt.setDate(17, Date.valueOf("2019-07-16"));
+        if (!isNamespaceMapped) {
+            upsertStmt.setString(18, TENANT1);
+        }
+        upsertStmt.executeUpdate();
+        conn2.commit();
+
+        String select = "SELECT INT1 FROM " + childViewName
```

Comment:
@gokceni @kadirozde  Shouldn't this be checking whatever the last column of the view PK is, to see if it's been truncated or not? In some of your test cases calling this method, INT1 is not the end of the view PK. 

I'd expect cases where, say, it ends in a text field to have a String whose ending is either SEPARATOR_BYTE or DESC_SEPARATOR_BYTE as appropriate and verify that the String isn't truncated. Or have I misunderstood the purpose of the test?

Reply:
Yes, whatever the last PK column is, we need to check its value. I missed that we only check INT1 here.  Good catch! The test method can take extra input for the last PK column and expected value for it and verify it.

Reply:
Check it out. Most of the time, they were all ending with INT1. There is a couple of double and text too.